presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroColumnDecoder.java

```diff
@@ -100,6 +103,10 @@ private boolean isSupportedPrimitive(Type type)
                         BigintType.BIGINT,
                         BooleanType.BOOLEAN,
                         DoubleType.DOUBLE,
+                        IntegerType.INTEGER,
+                        RealType.REAL,
+                        SmallintType.SMALLINT,
+                        TinyintType.TINYINT,
```

Comment:
add integer types together, in order: tiny, small, int, biging

add real before double

make the list an ImmutableSet and make it a class constant